cpp/tests/reshape/interleave_columns_tests.cu

```diff
@@ -176,4 +177,173 @@ TYPED_TEST(InterleaveColumnsTest, MismatchedDtypes)
+                                                 {false, false, true, true, true, true});
+  auto col1 = cudf::test::strings_column_wrapper({"", "valid", "null", "null", "valid", ""},
+                                                 {true, true, false, false, true, true});
+  auto col2 = cudf::test::strings_column_wrapper({"valid", "", "valid", "", "null", "null"},
+                                                 {true, true, true, true, false, false});
+
+  auto exp_results = cudf::test::strings_column_wrapper({"null",
+                                                         "",
+                                                         "valid",
+                                                         "null",
+                                                         "valid",
+                                                         "",
+                                                         "",
+                                                         "null",
+                                                         "valid",
+                                                         "valid",
+                                                         "null",
+                                                         "",
+                                                         "",
+                                                         "valid",
+                                                         "null",
+                                                         "valid",
+                                                         "",
+                                                         "null"},
+                                                        {false,
+                                                         true,
+                                                         true,
+                                                         false,
+                                                         true,
+                                                         true,
+                                                         true,
+                                                         false,
+                                                         true,
+                                                         true,
+                                                         false,
+                                                         true,
+                                                         true,
+                                                         true,
+                                                         false,
+                                                         true,
+                                                         true,
+                                                         false});
+
+  auto results = cudf::interleave_columns(cudf::table_view{{col0, col1, col2}});
+  cudf::test::expect_columns_equal(*results, exp_results, true);
```

Comment:
This is probably unnecessary for CI builds (granted, it probably won't error to begin with).

```suggestion
  cudf::test::expect_columns_equal(*results, exp_results);
```

Reply:
i'd this to know everything that failed and not just the first failure, in case the test fails somewhere.